src/components/Typography/Typography.tsx

```diff
@@ -29,15 +31,45 @@ const DynamicText = ({
     variant = 'body16',
     italic,
     monospace,
+    copyable,
+    children,
     ...otherProps
 }: TypographyProps) => {
     const Tag = getTag(variant);
-    // @ts-ignore
-    return <Tag {...otherProps} />;
+    const [isCopied, setIsCopied] = useState(false);
+
+    const copyToClipboard = (): void => {
+        if (typeof navigator == 'undefined') return;
+        navigator.clipboard.writeText(`${children}`);
+        setIsCopied(true);
+    };
+
+    const renderCopyIcon = () =>
+        copyable && (
+            <CopyIconStyled
```

Comment:
this is nearly perfect and a great PR... all my cooments are right on this line

1. the naming convention is not aligned here... a styled button should be `ButtonStyled` and if you had one already, this could be `ButtonStyledCopy`
2. We should really use <Button /> props: icon, small, transparent BUT i do see that the icon changes so....
3. Really this `<CopyButton />` needs to be its own compoents (its used in a few places in admin and now twice in our kit

BUT, lets not block the PR, can you please open a task to make `<CopyButton />` and replace it here and in `<Input />`

Reply:
I made the issue 153 so you can disregard point 2 and 3, but could you take a quick pass at the naming of the styled elements in this component please?

https://github.com/web3ui/web3uikit/issues/153